__device-tests__/pages/editor-page.js

```diff
@@ -270,4 +271,40 @@ export default class EditorPage {
 		const text = await this.getTextForListBlock( block );
 		return text.toString();
 	}
+
+	// =========================
+	// Heading Block functions
+	// =========================
+	async addNewHeadingBlock() {
+		await this.addNewBlock( this.headingBlockName );
+	}
+
+	async getHeadingBlockAtPosition( position: number ) {
+		return this.getBlockAtPosition( position, this.headingBlockName );
+	}
+
+	// Inner element changes on iOS if Heading Block is empty
+	async getTextViewForHeadingBlock( block: wd.PromiseChainWebdriver.Element, empty: boolean ) {
+		let textViewElementName = empty ? 'XCUIElementTypeStaticText' : 'XCUIElementTypeTextView';
+		if ( isAndroid() ) {
+			textViewElementName = 'android.widget.EditText';
+		}
+
+		const accessibilityId = await block.getAttribute( this.accessibilityIdKey );
+		const blockLocator = `//*[@${ this.accessibilityIdXPathAttrib }="${ accessibilityId }"]//${ textViewElementName }`;
+		return await this.driver.elementByXPath( blockLocator );
+	}
+
+	// Assumes Heading Block is empty
+	async sendTextToHeadingBlock( block: wd.PromiseChainWebdriver.Element, text: string ) {
+		const textViewElement = await this.getTextViewForHeadingBlock( block, true );
+		return await typeString( this.driver, textViewElement, text );
+	}
+
+	// Fails if Heading block is not empty
```

Comment:
Should this say that this fails if the Heading Block "is empty" instead of "is not empty"?

If this does need to be updated, would it make sense to make the comment for `sendTextToHeadingBlock` follow the same pattern as the comment we use here? For example, they could both be either "Fails if Heading Block is/is-not empty" or they could both have the form of "Assumes Heading Block is empty/not-empty". It wouldn't make sense to do this if both methods don't (for example) fail when they receive the wrong input, but if it does make sense, having a shared form would make the behavior a bit easier to understand quickly. I do not feel remotely strongly about this, just a thought.

Reply:
The comment is correct, when the heading block is not empty the inner element changes and then the current code won't work if that's the case. But I get what you're saying with the consistency, I've made that change 👍

Reply:
Thanks for explaining! Let me ask another question to help me understand.

It's not intuitive to me that a `getText...` method like this would only work on empty blocks. Would that not result in the method just always return an empty string (since the block is "empty")? I may just be misunderstanding what we mean by empty here.

This is probably related, but I'm also a bit confused by the `empty` value in the calls to `getTextViewForHeadingBlock in both `sendTextToHeadingBlock` and this `getTextForHeadingBlock`. It sounds like we're assuming the Heading Block is empty in both of these methods, but we're passing different `empty` boolean values to `getTextViewForHeadingBlock` in the two methods. If we're assuming that the Heading block is empty in both methods, I thought (probably mistakenly) that we would also pass `empty = true` to `getTextViewForHeadingBlock` in both methods. Can you help me understand where I'm going wrong?

Reply:
@mchowning you're right here! I think my mind was still thinking of an earlier state of the code, the code works as you understand it and you'd expect it to, my bad here! 🙇

Reply:
I've gone ahead and removed the misleading comments